Dockerfile

```diff
@@ -12,24 +12,25 @@ WORKDIR /app
 
 COPY poetry.toml poetry.lock pyproject.toml ./
 
-RUN poetry install --no-dev
+RUN poetry config virtualenvs.create false --local &&\
+    poetry install
 
 COPY . .
 
 FROM base as prod
-ENTRYPOINT poetry run gunicorn --bind 0.0.0.0:5000 "app:create_app(None)"
+CMD poetry run gunicorn --bind 0.0.0.0:$PORT "app:create_app(None)"
 
 FROM base as dev
-ENTRYPOINT poetry run flask run --host=0.0.0.0
+CMD poetry run flask run --host=0.0.0.0
 
 FROM base as test
 RUN poetry install
 
 FROM test as unit
-ENTRYPOINT poetry run pytest tests/unit
+CMD poetry run pytest tests/unit
 
 FROM test as integration
-ENTRYPOINT poetry run pytest tests/integration
+CMD poetry run pytest tests/integration
```

Comment:
It's a bit simpler to have a single test image and set the ENTRYPOINT to `poetry run pytest`

And then you can run e.g. `docker run image-name tests/unit` to specifically unit tests. But you have just one image and decide things with your run command instead.